tests/TaskLineRenderer.test.ts

```diff
@@ -135,6 +129,13 @@ describe('task line rendering', () => {
         // Check that eventually the correct text was rendered
         expect((internalDescriptionSpan as HTMLSpanElement).innerText).toEqual('This is a simple task');
     });
+});
+
+describe('task line rendering - global filter', () => {
+    afterEach(() => {
+        GlobalFilter.getInstance().reset();
+        GlobalFilter.getInstance().setRemoveGlobalFilter(false);
+    });
```

Comment:
Reviewing the individual commits, I see you've had to pay attention to which bits of global data need to be reset, and move the resets with the tests.

This feels error-prone to me.

I feel it would be saver to put the resetting code before the first `describe` block, that is, outside the `describe` blocks. Then every test is reset automatically.

```
afterEach(() => {
    GlobalFilter.getInstance().reset();
    GlobalFilter.getInstance().setRemoveGlobalFilter(false);
    resetSettings();
});
```

Reply:
Sorry, could you please elaborate what's the error-prone part here? If I comment out any of the `afterEach()` sections tests fail, so we need them for the tests to pass, which proves they are needed in the places where they are and not elsewhere.

Reply:
> Sorry, could you please elaborate what's the error-prone part here? If I comment out any of the `afterEach()` sections tests fail, so we need them for the tests to pass, which proves they are needed in the places where they are and not elsewhere.

Sure, good question. I'm not worried about spotting tests that fail due to some missing instruction - we will always spot failures.

I was more thinking about future tests being written that may spuriously pass, due to left-over data from the previous test.

Reply:
It’s not a big deal, and I’m happy to merge the PR as is, if you prefer.

Reply:
> I was more thinking about future tests being written that may spuriously pass, due to left-over data from the previous test.

Oh right, this makes perfect sense, let me push the updates. Do you want to rebase on top of the latest main as well?

Reply:
Thank you for being understanding. As you’re changing it any, if there are no conflicts a rebase would be much appreciated.

Reply:
Done. I tried commenting out each line separately in the new `afterEach()` and I have to note that removing `GlobalFilter.getInstance().setRemoveGlobalFilter(false);` doesn't change the test results. I'm not sure whether anything has to be done here.

Reply:
Interesting. Does that mean it is a test case that is not covered and should be? Or is the rendering behaviour of that option covered somewhere else?

Reply:
Ah no, it might mean that there is only one test case for it, and that test was being a good citizen and tidying up after itself.

Reply:
Will merge when I am next in front of my Mac.